src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,372 @@
+        index = 0  # keeps track of position in tensor components
+        # iterate over elements
+        for i, extent in enumerate(extents):
+            source_mesh = Mesh[dim](extent, basis_from_string(bases[i]),
+                                    quadrature_from_string(quadratures[i]))
+
+            interpolant = RegularGrid[dim](source_mesh, target_mesh)
+            grid_points = source_mesh.number_of_grid_points()
+
+            tensor_comps = []
+            # iterate over tensors
+            for j, tensor in enumerate(tensors):
+                dataVector = DataVector(tensor[index:index + grid_points],
+                                        copy=False)
+                interpolated_tensor = interpolant.interpolate(dataVector)
+                tensor_path = os.path.join(grid_names[i], tensor_names[j])
+                tensor_comps.append(
+                    TensorComponent(
+                        tensor_path, DataVector(interpolated_tensor,
+                                                copy=False)))
+
+            volume_data.append(
+                ElementVolumeData(target_mesh.extents(), tensor_comps,
+                                  target_mesh.basis(),
+                                  target_mesh.quadrature()))
+            index += grid_points
+
+        assert index == len(tensors[0])
+        target_vol.write_volume_data(obs,
+                                     source_vol.get_observation_value(obs),
+                                     volume_data)
+
+    source_file.close()
+    target_file.close()
+
+
+def _parse_args(sys_args):
+    """
+    defines and parses the command line arguments
+
+    Parameters
+    ----------
+    sys_args arguments passed via command line
+
+    Returns
+    -------
+    A dictionary of the parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        "Interpolate the tensor components of h5 files to a desired grid")
+
+    parser.add_argument("source_files",
+                        action="extend",
+                        type=lambda path: glob.glob(path),
+                        nargs='+',
+                        help="The path to one or multiple source .h5 files. "
+                        "Accepts glob wildcards.")
+
+    parser.add_argument(
+        "--target_files",
+        type=str,
+        nargs='+',
+        help="the path to where the interpolated function is written. "
+        "The number of target files must match the number of source files."
+        " When no target file is specified, the interpolated data "
+        "is written to the source file in a new volume data.")
+
+    parser.add_argument(
+        "--tensor_components",
+        type=str,
+        nargs='*',
+        help="The names of the tensors that are to be interpolated. "
+        "Accepts regular expression. "
+        "If none are specified, all tensors are interpolated.")
+
+    parser.add_argument("--grid_extents",
```

Comment:
rename `grid` -> `target` for these three target-mesh CLI args? I.e. `--target-extents --target-basis --target-quadrature`